File: dehaze-python/app/models/schema/order.py
from pydantic import BaseModel, Field


class OrderCreateForm(BaseModel):
    packageId: int = Field(..., description="套餐ID")
    couponId: int | None = Field(default=None, description="用户优惠券实例ID")
    payMethod: str = Field(..., description="支付方式")
    balanceAmount: int | None = Field(
        default=None, ge=0, description="组合支付时余额部分金额(分)"
    )


class PayRequest(BaseModel):
    payMethod: str = Field(..., description="支付方式")
    balanceAmount: int | None = Field(
        default=None, ge=0, description="组合支付时余额部分金额(分)"
    )


class RefundApplyForm(BaseModel):
    reasonType: str = Field(..., description="售后原因类型(after_sale/force_majeure/merchant/other)")
    reason: str | None = Field(default=None, description="退款原因说明")
    customReason: str | None = Field(default=None, description="自定义补充说明")


class RefundAuditForm(BaseModel):
    approved: bool = Field(..., description="是否通过")
    remark: str = Field(..., description="审核备注")


class AutoRenewConfigForm(BaseModel):
    packageId: int = Field(..., description="套餐ID")
    payMethod: str = Field(..., description="支付方式")
    enabled: bool = Field(..., description="是否启用")


class BalanceRefundForm(BaseModel):
    orderId: int | None = Field(default=None, description="关联订单ID(可空)")
    amount: int | None = Field(default=None, ge=0, description="退款金额(分,为空时按可用余额)")

File: dehaze-go/internal/model/bo/order.go
package bo

type OrderCreateForm struct {
	PackageID     int64  `json:"packageId"`
	CouponID      *int64 `json:"couponId"`
	PayMethod     string `json:"payMethod"`
	BalanceAmount *int64 `json:"balanceAmount"`
}

type PayRequest struct {
	PayMethod     string `json:"payMethod"`
	Channel       string `json:"channel"`
	BalanceAmount *int64 `json:"balanceAmount"`
}

type RefundApplyForm struct {
	ReasonType   string `json:"reasonType" binding:"required,oneof=after_sale force_majeure merchant other"`
	CustomReason string `json:"customReason"`
}

type RefundAuditForm struct {
	Approved bool   `json:"approved"`
	Remark   string `json:"remark"`
}

type BalanceRefundForm struct {
	OrderID *int64 `json:"orderId"`
	Amount  *int64 `json:"amount" binding:"omitempty,gte=0"`
}

type BalanceRefundAuditForm struct {
	Remark  string `json:"remark"`
	Channel string `json:"channel" binding:"omitempty,oneof=wechat alipay"`
}

type RechargeCreateForm struct {
	Amount    int64  `json:"amount" binding:"required,gt=0"`
	PayMethod string `json:"payMethod" binding:"required,oneof=wechat alipay"`
}

type AutoRenewConfigForm struct {
	PackageID int64  `json:"packageId"`
	PayMethod string `json:"payMethod"`
	Enabled   bool   `json:"enabled"`
}
